### DISCUSSION TITLE
Handle transaction rejected by user "User denied transaction signature"

### DISCUSSION THREAD

**Discussion by hamzamughal74** [wevm/wagmi#1906] [Q&A] [answered]
Hi i'm using usePrepareContractWrite, useContractWrite and useWaitForTransaction 
user is connected to metamask through walletConnect Modal 
when he rejects transaction from metamask popup box , no of above hooks catching this "User denied transaction signature" error
any solution please?

**Comment by SaadNiaziDev** [comment, marked as answer]
use onError callback in useContractWrite

https://wagmi.sh/react/prepare-hooks/usePrepareContractWrite#onerror-optional
